test/sql/CMakeLists.txt

```diff
@@ -134,10 +154,37 @@ endforeach(TEMPLATE_FILE)
 
 list(SORT TEST_FILES)
 file(REMOVE ${TEST_SCHEDULE})
+set(PARALLEL_GROUP_SIZE 20)
+
+# append solo tests to schedule first
+foreach(TEST_FILE ${TEST_FILES})
+  string(REGEX REPLACE "(.+)\.sql" "\\1" TESTS_TO_RUN ${TEST_FILE})
+  if(TESTS_TO_RUN IN_LIST SOLO_TESTS)
+    file(APPEND ${TEST_SCHEDULE} "test: ${TESTS_TO_RUN}\n")
+  endif()
+endforeach(TEST_FILE)
+
+set(GROUP_MEMBERS 0)
+foreach(TEST_FILE ${TEST_FILES})
+  string(REGEX REPLACE "(.+)\.sql" "\\1" TESTS_TO_RUN ${TEST_FILE})
+  if((NOT (TESTS_TO_RUN IN_LIST SOLO_TESTS)) AND (NOT (TESTS_TO_RUN IN_LIST POST_TESTS)))
+    math(EXPR GROUP_MEMBERS ${GROUP_MEMBERS}%${PARALLEL_GROUP_SIZE})
+    if(GROUP_MEMBERS EQUAL 0)
+      file(APPEND ${TEST_SCHEDULE} "\ntest: ")
+    endif()
+    file(APPEND ${TEST_SCHEDULE} "${TESTS_TO_RUN} ")
+    MATH(EXPR GROUP_MEMBERS ${GROUP_MEMBERS}+1)
```

Comment:
why not just do the mod ${PARALLEL_GROUP_SIZE} here instead of at line 171. Then you can delete line 171.